### PR TITLE
txn: disable rc read check ts for cursor read mode

### LINKED ISSUE #42184: txn: disable rc read check ts for prepared statement cursor read mode
## Feature Request

**Is your feature request related to a problem? Please describe:**

The [`tidb_rc_read_check_ts`](https://docs.pingcap.com/zh/tidb/dev/system-variables#tidb_rc_read_check_ts-%E4%BB%8E-v600-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5) could report errors if the returned result set is large, especially when cursor read is used.

**Describe the feature you'd like:**

Diable this feature when cursor read is used.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve?
Disable the `RCReadCheckTS` usage for prepared statement cursor read mode.

Issue Number: close #42184 

Problem Summary:

### Check List

Tests 

- [x] Unit test

Side effects

Documentation

- [x] Affects user behaviors
- [x] Contains variable changes

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Disable the rc read check ts for prepared statement cursor read mode.
```

### CODE REVIEW

**Review by ekexium — APPROVED** [review]
\

**Review by zyguan — APPROVED** [review]

File: store/copr/coprocessor.go [hunk 1]
```diff
@@ -858,6 +858,9 @@ func (sender *copIteratorTaskSender) run() {
 		if exit {
 			break
 		}
+		if connID > 0 {
```
Comment (you06) [inline store/copr/coprocessor.go:861]: Can use `RequestSource` to specify paused cop tasks instead?

Comment (cfzjywxk) [reply store/copr/coprocessor.go:861]: It should be ok,  adding `connID` is helpful to group requests by session in debug logs.

Comment (you06) [reply store/copr/coprocessor.go:861]: OK, LGTM

**Pull request merged**